csrc/scheduler/matmul_heuristic_plugin.h

```diff
@@ -0,0 +1,77 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <mma_type.h>
+#include <scheduler/matmul_heuristic.h>
+#include <scheduler/mma_utils.h>
+
+#include <optional>
+
+namespace nvfuser {
+
+namespace matmul_heuristic_plugin {
+
+//! This is the information available to the plugin to determine the kernel
+//! configuration. For API stability, these should not be accessed directly when
+//! implementing plugins, but rather through these accessors.
+//! matmul_heuristic_plugin_api.h
+struct NVF_API ProblemDescription;
```

Comment:
Why are we duplicating code in `matmul_heuristic_plugin_api.h`? Is it for `NVF_API`? Does it has to be `NVF_API`?

Reply:
Yes. It needs to be NVF_API and it's convenient to keep the includes minimal in the external header. I tried to be really conservative at first but I could probably include `visibility.h` without problems and just make these `NVF_API` in the external header. I'll try it.

Reply:
If it's not NVF_API then we can have missing symbols when importing the plugin. I hit this in an early version.

Reply:
Code is no longer duplicated. Plugins subclass a class that defines the problem and config.